src/operators.jl

```diff
@@ -244,65 +267,161 @@ function flatten(input::Signal; typ=Any, name=auto_name!("flatten", input))
 end
 
 function connect_flatten(output, input)
-    let current_node = value(input),
-        callback = (output, timestep) -> begin
-            send_value!(output, value(value(input)), timestep)
+    let current_node = value(input)
+        wire_flatten(current_node, subtree_actions::Vector{Action}) = begin
+            #children of this flatten node need to know to update
+            #on changes to the input sigsig (allroots(input)),
+            #or changes to the value of the current sig (roots == allroots(current_node))
+            # @show "switched to " current_node
+            roots = allroots(current_node)
+            for root in roots
+                #ensure updates when current_node gets pushed a new value
+                add_action!(output, root) do output
+                    send_value!(output, value(current_node))
+                end
+                #ensure updates for subtree too, if action is already in queue
+                #remove it and readd it so it's deeper down. XXX I think that's
+                #mostly correct but prob not 100%
+                queue = action_queues[root]
+                for action in subtree_actions
+                    deleteat!(queue, findin(queue, 6))
```

Comment:
I think it's supposed to be `action`. I guess it worked but would have caused some things to be re-run twice since in some cases since they weren't removed properly.